tangelo/linq/target/target_qulacs.py

```diff
@@ -74,36 +77,85 @@ def simulate_circuit(self, source_circuit: Circuit, return_statevector=False, in
         if initial_statevector is not None:
             state.load(initial_statevector)
 
-        # If you don't want to save the mid-circuit measurements for a mixed state
-        if (source_circuit.is_mixed_state or self._noise_model) and not save_mid_circuit_meas:
+        if desired_meas_result is not None:
+            n_meas = source_circuit.counts.get("MEASURE", 0)
+            self._current_state = None
+            n_attempts = 0
+            n_success = 0
+            n_shots = self.n_shots if self.n_shots is not None else -1
+            # Permit 0.1% probability events
```

Comment:
Skip a line.

Don't hesitate to throw a comment or two to avoid ending up with monolithic code, or skip a line somewhere if you think it helps parsing the main conditional branches. This applies to everything in this file.